rasa/cli/x.py

```diff
@@ -291,16 +289,77 @@ def rasa_x(args: argparse.Namespace):
         run_locally(args)
 
 
+async def _pull_runtime_config_from_server(
+    config_endpoint: Optional[Text],
+    attempts: int = 30,
+    wait_time_between_pulls: Union[int, float] = 0.5,
+    keys: Iterable[Text] = ("endpoints", "credentials"),
+) -> Optional[List[Text]]:
+    """Pull runtime config from `config_endpoint`.
+
+    Returns a path to yaml dumps, each containing the contents of one of `keys`.
+    """
+
+    while attempts > 0:
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(config_endpoint) as resp:
+                    if resp.status == 200:
+                        rjs = await resp.json()
+                        try:
+                            return [
+                                io_utils.dump_dict_to_temporary_yaml_file(rjs[k])
+                                for k in keys
+                            ]
+                        except KeyError as e:
+                            cli_utils.print_error_and_exit(
+                                "Failed to find key '{}' in runtime config. "
+                                "Exiting.".format(e)
+                            )
+                    else:
+                        logger.debug(
+                            "Failed to get a proper response from remote "
+                            "server. Status Code: {}. Response: '{}'"
+                            "".format(resp.status, await resp.text())
+                        )
+        except aiohttp.ClientError as e:
+            logger.debug("Failed to connect to server. Retrying. {}".format(e))
+
+        await asyncio.sleep(wait_time_between_pulls)
+        attempts -= 1
+
+    cli_utils.print_error_and_exit(
+        "Could not fetch runtime config from server at '{}'. "
+        "Exiting.".format(config_endpoint)
+    )
+
+
 def run_in_production(args: argparse.Namespace):
     from rasa.cli.utils import print_success
-    from rasa.core.utils import AvailableEndpoints
 
     print_success("Starting Rasa X in production mode... 🚀")
-    args.endpoints = get_validated_path(
-        args.endpoints, "endpoints", DEFAULT_ENDPOINTS_PATH, True
-    )
-    endpoints = AvailableEndpoints.read_endpoints(args.endpoints)
-    _rasa_service(args, endpoints)
+
+    credentials_path, endpoints_path = _get_credentials_and_endpoints_paths(args)
+    endpoints = AvailableEndpoints.read_endpoints(endpoints_path)
+
+    _rasa_service(args, endpoints, None, credentials_path)
+
+
+def _get_credentials_and_endpoints_paths(args) -> Tuple[Optional[Text], Optional[Text]]:
```

Comment:
this will probably destroy your black formatting :-D 
```suggestion
def _get_credentials_and_endpoints_paths(args: argparse.Namespace) -> Tuple[Optional[Text], Optional[Text]]:
```